Add default-true parameter "~enable_odom_tf"

bmagyar opened the pull request:
https://github.com/ros-controls/ros_controllers/issues/118

bmagyar committed:
Add default-true parameter "~enable_odom_tf"

adolfo-rt commented:
Could the test suite be expanded to make sure that with this setting no tf data is being published?

bmagyar commented:
I'll cook something up and add to the PR.

adolfo-rt commented:
Once that's done we can merge. Thanks for the quick response!

bmagyar commented:
I added a new test to check that there is no frame when the tf odom is not enabled. I also added an extra test function to the general test to check that it is there in the default case.

adolfo-rt commented on diff_drive_controller/test/diff_drive_odom_tf_test.cpp:
```diff
@@ -0,0 +1,60 @@
+  }
+  // set up tf listener
+  tf::TransformListener listener;
+  // wait for 5s
```
The comment is out of sync with the line below (5s vs 2s). Maybe we can just remove it

bmagyar replied:
done

adolfo-rt commented on diff_drive_controller/test/diff_drive_test.cpp:
```diff
@@ -121,6 +122,21 @@ TEST_F(DiffDriveControllerTest, testTurn)
+  }
+  // set up tf listener
+  tf::TransformListener listener;
+  // wait for 5s
```
Same here

bmagyar replied:
done

adolfo-rt commented:
Minor comments inline with the code.

efernandez commented:
LGTM :+1:

adolfo-rt commented:
+1. merging

mikepurvis commented:
Thanks again!

bmagyar commented:
:+1:
